optaplanner-examples/src/main/java/org/optaplanner/examples/pas/domain/BedDesignation.java

```diff
@@ -118,4 +118,49 @@ public String toString() {
         return admissionPart.toString();
     }
 
+    //Testing builder
+
+    public BedDesignation build(){
+        admissionPart = new AdmissionPart();
+        bed = new Bed();
+        return this;
+    }
+
+    //Testing methods
+
+    public BedDesignation withNights(int first, int last) {
+        Night firstNight = new Night();
+        firstNight.setIndex(first);
+        admissionPart.setFirstNight(firstNight);
+        Night lastNight = new Night();
+        lastNight.setIndex(last);
+        admissionPart.setLastNight(lastNight);
+        return this;
+    }
+
+    public BedDesignation withId(Long id) {
+        this.id = id;
+        return this;
+    }
+
+    public BedDesignation withPatient(Patient patient) {
+        this.admissionPart.setPatient(patient);
+        return this;
+    }
+
+    public BedDesignation withBed(Bed bed) {
+        this.bed = bed;
+        return this;
+    }
+
+    public BedDesignation withRoom(Room room) {
+        this.bed.setRoom(room);
+        return this;
+    }
+
+    public BedDesignation withSpecialism(Specialism spec1) {
+        this.admissionPart.setSpecialism(spec1);
+        return this;
+    }
```

Comment:
In my original comment, I meant that these methods would simply become setters. That is `public void set...(...)`.
This way, it becomes consistent with the other examples. (Although unfortunately the test code gets a bit uglier.)

Sorry for not being clear earlier.

Reply:
https://github.com/kiegroup/optaplanner/pull/872/commits/1f9f2b888293c0e1f544dfc79931c17386a57b41